Lib/pathlib.py

```diff
@@ -425,9 +337,9 @@ def __init__(self, pat, child_parts, flavour):
         self.match = flavour.compile_pattern(pat)
         _Selector.__init__(self, child_parts, flavour)
 
-    def _select_from(self, parent_path, is_dir, exists, scandir):
+    def _select_from(self, parent_path, is_dir, exists):
         try:
-            with scandir(parent_path) as scandir_it:
+            with parent_path._scandir() as scandir_it:
```

Comment:
Why not use just `os.scandir()`?

Reply:
For bpo-24132, all filesystem access needs to happen in `Path` so that it can be customized in subclasses of a future `AbstractPath`.

(This is also why I've ensured that `os` functions are called from only one place in `Path`).

Reply:
Could you add a comment in the code about that?
I read this before then forgot and today didn’t see why we had `os.getcwd()` called directly in some places but the `self._scandir` apparently needless indirection!

Reply:
I've added a comment!

Reply:
I was thinking about putting the comment in next to the _scandir definition!

Reply:
Ah, sorry. I've moved the comment, hope it looks OK?